tests/CMakeLists.txt

```diff
@@ -1,11 +1,12 @@
 
+include(CTest)
```

Comment:
I'm surprised we don't have this.  Looking through the code it seems we were getting it accidentally from `taskflow`.  Thanks for adding it.

It should be in the top-level CMake file, after the call to `project`, but before we start processing anything (ie: before any `add_subdirectory`).  So between lines 37 and 259 if my eyes are right.

Reply:
Thanks! I've adjusted this to be in the top-level CMake file now.

Reply:
What was the motivation for moving this?

Does this impact CI?

Do we use ctest in CI?

I ask, because I thought that the level where you call this impacts where ctest files are generated.

Also, I’m used to seeing this gated by BUILD_TESTS=ON

Reply:
I just did a quick grep, and looks like we don't use ctest at all in any scripts or workflows.

We directly invoke test executables.

i.e.

```
./tests/scripts/tg/run_tg_unit_tests.sh:  TT_METAL_ENABLE_REMOTE_CHIP=1 ./build/test/tt_metal/unit_tests_fast_dispatch --gtest_filter="CommandQueueSingleCardFixture.*"
```

This means putting it at the top level shouldn't break anything.